src/scorers.rs

```diff
@@ -283,6 +283,115 @@ impl ScorerBuilder for SumOfScorersBuilder {
     }
 }
 
+/**
+Composite Scorer that takes any number of other Scorers and returns the product of their [`Score`]. If the resulting score
+is less than the threshold, it returns 0.
+
+The Scorer also provides a compensation factor based on the number of Scores passed to it. This can be disabled by passing
+`false` to the `build` argument.
+
+### Example
+
+```ignore
+Thinker::build(0.5, true)
+    .when(
+        ProductOfScorers::build()
+          .push(MyScorer)
+          .push(MyOtherScorer),
+        MyAction::build());
+```
+ */
+
+#[derive(Component, Debug)]
+pub struct ProductOfScorers {
+    threshold: f32,
+    apply_compensation: bool,
+    scorers: Vec<ScorerEnt>,
+}
+
+impl ProductOfScorers {
+    pub fn build(threshold: f32, apply_compensation: bool) -> ProductOfScorersBuilder {
```

Comment:
Can you move `apply_compensation` to the Builder and pick a default?

So you would be able to do this:

```rust
ProductOfScorers::build()
  .apply_compensation(false)
  .push(MyScorer)
  .push(MyOtherScorer)
```

Reply:
Ah, this is a better approach thanks.